From ecashin@coraid.com Thu Jan 19 11:11:56 2006
Message-ID: <6708136ac02e66a47b3c32b744cd5ca1@coraid.com>
From: "Ed L. Cashin" <ecashin@coraid.com>
Date: Thu, 19 Jan 2006 13:46:27 -0500
CC: ecashin@coraid.com, Greg K-H <greg@kroah.com>
Subject: [PATCH 2.6.15-git9] aoe [6/8]: update device information on last close

Instead of making the user wait or do it manually, refresh
device information on its last close by issuing a config
query to the device.

Signed-off-by: "Ed L. Cashin" <ecashin@coraid.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/block/aoe/aoeblk.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- gregkh-2.6.orig/drivers/block/aoe/aoeblk.c
+++ gregkh-2.6/drivers/block/aoe/aoeblk.c
@@ -109,7 +109,7 @@ aoeblk_release(struct inode *inode, stru
 
 	spin_lock_irqsave(&d->lock, flags);
 
-	if (--d->nopen == 0 && !(d->flags & DEVFL_UP)) {
+	if (--d->nopen == 0) {
 		spin_unlock_irqrestore(&d->lock, flags);
 		aoecmd_cfg(d->aoemajor, d->aoeminor);
 		return 0;
